runtime/src/main/java/io/quarkiverse/fury/FuryRecorder.java

```diff
@@ -92,10 +81,31 @@ public void registerClass(
             fury.register(clazz, (short) classId);
         } else {
             // Generate serializer bytecode.
-            fury.register(clazz, serializer == null);
+            if (config.requireClassRegistration()) {
+                fury.register(clazz, serializer == null);
+            }
         }
         if (serializer != null) {
             fury.registerSerializer(clazz, serializer);
+        } else {
+            // Generate serializer bytecode.
+            try {
+                Method createSerializerAhead = ClassResolver.class.getDeclaredMethod(
+                        "createSerializerAhead", Class.class);
+                createSerializerAhead.setAccessible(true);
```

Comment:
Well, it seem that `createSerializerAhead` is a internal method, right? not sure if it could be public in the future. I think we should keep only to use the Fury public API because this internal method could be changed or removed without any compatible concern.

Reply:
OK, that's good. And can you also update the doc section with these changes? IIUC, if `quarkus.fury.required-class-registration=false`, no need to register classes both on client and server side, but there might be some security concerns when especially deserilazing from the untrust sources. User should know this trade-off and make sure to enable `required-class-registration` or not.

And after that I think it will be good to release `0.2.0` soon, thanks a lot!

Reply:
I added a class checker, which will have same security level as class registration enabled.